import.go

```diff
@@ -44,19 +65,97 @@ func newImporter(tree *MutableTree, version int64) (*Importer, error) {
 		return nil, errors.New("tree must be empty")
 	}
 
-	return &Importer{
-		tree:    tree,
-		version: version,
-		batch:   tree.ndb.db.NewBatch(),
-		stack:   make([]*Node, 0, 8),
-	}, nil
+	importer := &Importer{
+		tree:             tree,
+		version:          version,
+		batch:            tree.ndb.db.NewBatch(),
+		stack:            make([]*Node, 0, 8),
+		batchMtx:         sync.RWMutex{},
+		desiredBatchSize: defaultDesiredBatchSize,
+		maxBatchSize:     defaultMaxBatchSize,
+		chNodeData:       make(chan NodeData, 2*defaultDesiredBatchSize),
+		chNodeDataWg:     sync.WaitGroup{},
+		chBatch:          make(chan db.Batch, 1),
+		chBatchWg:        sync.WaitGroup{},
+		chError:          make(chan error, 1),
+		allChannelClosed: false,
+	}
+
+	importer.chNodeDataWg.Add(1)
+	go setBatchData(importer)
+
+	importer.chBatchWg.Add(1)
+	go batchWrite(importer)
+
+	return importer, nil
+}
+
+// WithDesiredBatchSize set the desired batch size for write
+func (i *Importer) WithDesiredBatchSize(batchSize uint32) *Importer {
+	i.desiredBatchSize = batchSize
```

Comment:
The mutability of `desiredBatchSize` and `maxBatchSize` during an ongoing import seems like too much complexity for little gain.